service/SpaceCenter/src/Services/Orbit.cs

```diff
@@ -362,5 +362,138 @@ public double OrbitalSpeedAt (double time)
+        [KRPCMethod]
+        public double RadiusAt(double time)
+        {
+            return InternalOrbit.getRelativePositionAtUT(time).magnitude;
+        }
+
+        /// <summary>
+        /// The orbital radius at the given time, in meters.
+        /// </summary>
+        /// <param name="time">Time from now, in seconds.</param>
+        /// <param name="rf">Reference Frame.</param>
+        [KRPCMethod]
+        public Tuple3 PositionAt(double time, ReferenceFrame rf)
+        {
+            return rf.PositionFromWorldSpace(InternalOrbit.getPositionAtUT(time)).ToTuple();
+
+        }
+
+
+        /// <summary>
+        /// Estimates time of closest approach in the next orbit.
+        /// </summary>
+        /// <param name="target">Other vessel.</param>
+        [KRPCMethod]
+        public double ClosestApproach(Vessel target)
+        {           
+            Orbit b = target.Orbit;
+            double approachTime = Planetarium.GetUniversalTime();
+            double approachDistance = double.MaxValue;
+            double mintime = Planetarium.GetUniversalTime();
+            double interval = InternalOrbit.period;
+            if (InternalOrbit.eccentricity > 1.0) { interval = 100 / InternalOrbit.meanMotion; }
+            double maxtime = mintime + interval;
+
+            //Conduct coarse search
+            double timestep = (maxtime - mintime) / 20;
+            double placeholder = mintime;
+            while (placeholder < maxtime)  
+            {
+                Vector3d PosA = InternalOrbit.getPositionAtUT(placeholder);
+                Vector3d PosB = b.InternalOrbit.getPositionAtUT(placeholder);
+                double thisDistance = Vector3d.Distance(PosA, PosB);
+                if (thisDistance < approachDistance)
+                {
+                    approachDistance = thisDistance;
+                    approachTime = placeholder;
+                }
+                placeholder += timestep;
+            }
+
+            //Conduct fine search
+            double fine_mintime = approachTime - timestep;
+            double fine_maxtime = approachTime + timestep;
+            if (fine_maxtime > maxtime) fine_maxtime = maxtime;
+            if (fine_mintime < mintime) fine_mintime = mintime;
+            timestep = (fine_maxtime - fine_mintime) / 50;
+            placeholder = fine_mintime;
+
+            while (placeholder < fine_maxtime)
+            {
+                Vector3d PosA = InternalOrbit.getPositionAtUT(placeholder);
+                Vector3d PosB = b.InternalOrbit.getPositionAtUT(placeholder);
+                double thisDistance = Vector3d.Distance(PosA, PosB);
+                if (thisDistance < approachDistance)
+                {
+                    approachDistance = thisDistance;
+                    approachTime = placeholder;
+                }
+                placeholder += timestep;
+            }
+            return approachTime;
+        }
+
+
+        /// <summary>
```

Comment:
I think having these stats for several orbits would be very useful. I'd add a separate RPC though that returns a list - so that if you just want the stats for one orbit you don't have to unpack a list.

The hohmann code in the toolkit repo uses math to calculate the burn directly. Not sure why it would be off.

I think best to leave that to client code. I tend to avoid adding things like that as then I don't need to maintain even more code - krpc is pretty huge as it is! Also with things like this there is usually already a mod on ckan that has the required functionality - so would be easier to just add some RPCs to expose it through the server.